web/modules/custom/dpl_related_content/src/Services/RelatedContent.php

```diff
@@ -219,62 +233,122 @@ public function getContent(): array {
 
-    // We want to combine events and nodes into a single array, however,
-    // we want to interlace them, so it becomes mixed, rather than just
-    // all the nodes and then all the events.
-    // We could also sort the merging based on the actual date values, but this
-    // will be too expensive for what it is worth.
-    $content = [];
-
     // Remove array keys, so we can do the sorting.
     $events = array_values($events);
     $nodes = array_values($nodes);
 
-    $node_view_builder = $this->entityTypeManager->getViewBuilder('node');
     $event_view_builder = $this->entityTypeManager->getViewBuilder('eventinstance');
+    $node_view_builder = $this->entityTypeManager->getViewBuilder('node');
 
-    // We find the longer array, to do a simple FOR loop.
-    $length = max(count($events), count($nodes));
+    $node_length = count($nodes);
+    $event_length = count($events);
+    $max_length = max($node_length, $event_length);
+    $content = [];
 
-    for ($i = 0; $i < $length;) {
-      if ($i >= $this->maxItems) {
+    for ($i = 0; $i < $max_length; $i++) {
+      if (count($content) >= $this->maxItems) {
         break;
       }
 
-      if (isset($events[$i])) {
+      if ($i < $event_length) {
         $content[] = $event_view_builder->view($events[$i], $this->contentViewMode);
-        $i++;
       }
 
-      if ($i >= $this->maxItems) {
+      if (count($content) >= $this->maxItems) {
         break;
       }
 
-      if (isset($nodes[$i])) {
+      if ($i < $node_length) {
         $content[] = $node_view_builder->view($nodes[$i], $this->contentViewMode);
-        $i++;
       }
     }
 
-    return [
-      '#theme' => 'dpl_related_content',
-      '#title' => $this->title,
-      '#items' => $content,
-      '#list_style' => $this->listStyle,
-      // The results should be cached, but, they have a lot of dependencies -
-      // even depending on the current time (finding future events).
-      // The individual pieces of content are cached themselves, so for the full
-      // result list, we'll add an easily-invalidated cache, with a bunch of
-      // cache tags, and even a time-based cache.
-      '#cache' => [
-        // Max age of 12 hours.
-        '#max-age' => (60 * 60 * 12),
-        '#tags' => ['eventinstance_list', 'eventseries_list', 'node_list'],
-      ],
-    ];
+    return $content;
+  }
+
+  /**
+   * Getting a combined condition group, taking into account AND/OR logic.
+   *
+   * @param \Drupal\Core\Entity\Query\QueryInterface|SelectInterface $query
+   *   The query on which we should create the condition group.
+   * @param array<mixed> $filters
+   *   Array of field_keys => value, that we want to look up.
+   */
+  private function addFilterConditions(QueryInterface|SelectInterface $query, array $filters): void {
+    // Get rid of any empty filters.
+    $filters = array_filter($filters);
+
+    if (empty($filters)) {
+      return;
+    }
```

Comment:
Is this necessary? If the array is empty then the subsequent loop will not run.

Reply:
yep, because after the loop we do this:

```
    if ($or_group) {
      $query->condition($or_group);
    }
```

and `$or_group` is set before the loop, so it would exist.